src/core/server/http/cookie_session_storage.ts

```diff
@@ -100,6 +106,12 @@ class ScopedCookieSessionStorage<T extends Record<string, any>> implements Sessi
   }
 }
 
+function validateOptions(options: SessionStorageCookieOptions<any>) {
+  if (options.sameSite === 'None' && options.isSecure !== true) {
```

Comment:
We might move this validation into the Security plugin to fail fast. However, this logic doesn't belong to the plugin. @elastic/kibana-security should I move it into the plugin?

Reply:
That makes sense to me, I'm fine to leave in core